File: FidelityHub.Application/Helpers/Api/RequestResult.cs
using System;
using System.Net;

namespace FidelityHub.Application.Helpers.Api
{
    public class RequestResult
    {
        public string Message { get; set; }
        public HttpStatusCode Code { get; set; }
        public DateTime Timestamp { get; set; }
        public object Data { get; set; }

        public RequestResult(HttpStatusCode code, string message, object model)
        {
            this.Code = code;
            this.Message = message;
            this.Timestamp = DateTime.Now;
            this.Data = model;
        }
    }
}

File: FidelityHub.Web/ClientApp/src/app/shared/models/requestResult.ts
export class RequestResult {
  code: number;
  message: string;
  timestamp: Date;
  data: any;

  constructor(code: number, message: string, data: any){
    this.timestamp = new Date();
    this.message = message;
    this.code = code;
    this.data = data;
  }
}
